http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala

```diff
@@ -167,11 +167,17 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
                 persistentDiskOpt <- rc.persistentDiskId.flatTraverse(did =>
                   persistentDiskQuery.getPersistentDiskRecord(did).transaction
                 )
-                _ <- persistentDiskOpt.traverse_(d =>
-                  googleDisk.deleteDisk(googleProject, rc.zone, d.name) >> persistentDiskQuery
-                    .updateStatus(d.id, DiskStatus.Deleted, ctx.now)
-                    .transaction
-                )
+                _ = persistentDiskOpt match {
+                  case Some(value) =>
+                    if (value.status == DiskStatus.Creating || value.status == DiskStatus.Failed) {
```

Comment:
Am I understanding correctly that we want to delete any disk in a non ready status? In that case, it might be easier to have the following condition:
`if (value.status != DiskStatus.Ready)`

Reply:
Regardless, adding a comment explaining the expected behavior would be nice :)

Reply:
Personally, I don't think we should affect the PD at all. But if we do want to, do we know what the 'Restoring' status is used for? I would think we don't want to delete a 'Restoring' disk as well.

Reply:
I like the idea of being explicit in the statuses we want to delete instead of the inverse. Because users often save disks between runtimes, I'd hate to eagerly delete.